clickhouse-http-client/src/main/java/com/clickhouse/client/http/HttpUrlConnectionImpl.java

```diff
@@ -101,11 +99,48 @@ private ClickHouseHttpResponse buildResponse(ClickHouseOutputStream output, Runn
                 displayName, queryId, summary, format, timeZone);
     }
 
+    private HttpURLConnection dispatchProxy(ClickHouseProxyType proxyType) throws IOException {
+        HttpURLConnection newConn = null;
+        ClickHouseConfig c = config;
+
+        switch (proxyType.toString()) {
+            case "HTTP":
+                if ( c.getProxyHostName() == "" || c.getProxyPort() == -1 ) {
+                    log.error("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort());
+                    throw new RuntimeException(String.format("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort()));
+                }
+                Proxy webProxy
+                        = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(c.getProxyHostName(), c.getProxyPort()));
+                newConn = (HttpURLConnection) new URL(url).openConnection(webProxy);
+                break;
+            case "DIRECT":
+                newConn = (HttpURLConnection) new URL(url).openConnection(Proxy.NO_PROXY);
+                break;
+            case "SOCKS":
+                if ( c.getProxyHostName() == "" || c.getProxyPort() == -1 ) {
+                    log.error("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort());
+                    throw new RuntimeException(String.format("config for HTTP proxy is invalid hostname [%s] port [%d]", c.getProxyHostName(), c.getProxyPort()));
+                }
+                Proxy socksProxy
+                        = new Proxy(Proxy.Type.SOCKS, new InetSocketAddress(c.getProxyHostName(), c.getProxyPort()));
+                newConn = (HttpURLConnection) new URL(url).openConnection(socksProxy);
+                break;
+            default:
+                // IGNORE
+                newConn = (HttpURLConnection) new URL(url).openConnection();
+        }
+        return newConn;
```

Comment:
Perhaps declare a `Proxy` variable before switch statement to replace most of `newConn = (HttpURLConnection) new URL(url).openConnection(...)`?